backend/ftp/ftp.go

```diff
@@ -868,11 +868,15 @@ func (f *Fs) mkdir(ctx context.Context, abspath string) error {
 	return err
 }
 
-// mkParentDir makes the parent of remote if necessary and any
-// directories above that
-func (f *Fs) mkParentDir(ctx context.Context, remote string) error {
-	parent := path.Dir(remote)
-	return f.mkdir(ctx, path.Join(f.root, parent))
+// mkParentDir makes the parent of the native path dirPath if
+// necessary and any directories above that
+func (f *Fs) mkParentDir(ctx context.Context, dirPath string) (err error) {
```

Comment:
Can we fix this without making `mkParentDir` take an absolute path. In my experience its best to keep the functions which take absolute paths down to a minimum.

Reply:
Do this internally in `Put` instead, you mean? I did something like that first, but then found that a `mkParentDirs` function exists in a few other backends already (webdav, jottacloud, mailru, yandex) - all of which had absolute path as argument.